### PR TITLE
Fix rasm2 endian -e flag

### PR DESCRIPTION
**Pull request by zamaudio**
See r2r # 385, test added

### CODE REVIEW

File: binr/rasm2/rasm2.c [hunk 1]
```diff
@@ -49,10 +49,10 @@ static int show_analinfo(const char *arg, ut64 offset) {
 	return ret;
 }
 
-static const char *has_esil(RAnal *a, const char *name) {
+static const char *has_esil(RAnal *lanal, const char *name) {
```
Comment (ret2libc) [inline binr/rasm2/rasm2.c]: Is `lanal` a typo?

File: binr/rasm2/rasm2.c [hunk 2]
```diff
@@ -49,10 +49,10 @@ static int show_analinfo(const char *arg, ut64 offset) {
...
 	RListIter *iter;
 	RAnalPlugin *h;
-	r_list_foreach (a->plugins, iter, h) {
+	r_list_foreach (anal->plugins, iter, h) {
```
Comment (radare) [inline binr/rasm2/rasm2.c]: why not just keep using `a`?

File: binr/rasm2/rasm2.c [hunk 3]
```diff
@@ -62,7 +62,7 @@ static const char *has_esil(RAnal *a, const char *name) {
 	return "__";
 }
 
-static void rasm2_list(RAsm *a, const char *arch) {
+static void rasm2_list(RAsm *la, const char *arch) {
```
Comment (radare) [inline binr/rasm2/rasm2.c]: why la

**Comment by radare** [comment]
Can you confirm this endian issue happens with pa and pad too?

File: binr/rasm2/rasm2.c (hunk 1 above)
Comment (zamaudio) [inline binr/rasm2/rasm2.c]: No, because "anal" is global, so i used lanal for local anal

File: binr/rasm2/rasm2.c (hunk 3 above)
Comment (zamaudio) [inline binr/rasm2/rasm2.c]: Because "a" is a global!!

File: binr/rasm2/rasm2.c (hunk 2 above)
Comment (zamaudio) [inline binr/rasm2/rasm2.c]: Because "a" is a global for RAsm

File: binr/rasm2/rasm2.c (hunk 3 above)
Comment (ret2libc) [inline binr/rasm2/rasm2.c]: @radare why do we have global variables here? Are they still useful?

**Comment by zamaudio** [comment]
In r2, there is no endian problem with pa or pad, it works as expected.  I don't think rasm2 has pa or pad.

**Comment by radare** [comment]
please update the pr

File: binr/rasm2/rasm2.c (hunk 1 above)
Comment (radare) [inline binr/rasm2/rasm2.c]: why not use `a`? like it was before

**Comment by radare** [comment]
i dont like that lanal thing, but at least it works. we will have time for cleaning code soon or late

**Comment by zamaudio** [comment]
Ok, do you like the globals removed?

**Comment by ret2libc** [comment]
As said in the CONTRIBUTING.md, globals are evil :P

**Pull request merged**